File: python_t3_full/Clone_571.py
def persistence(num) :
  num_str = str(num)
  count = 0
  while len(num_str) > 1:
    curr_num = 1
    for c in num_str:
        curr_num *= int(c)
    num_str = str(curr_num)
    count += 1
  return count


 def persistence(num):
  num_str = str(num)
  count = 0
  while len(num_str) > 1:
    prev_val, curr_val = 1, 1
    for x in num_str:
      prev_val *= curr_val
      curr_val = int(x)
      num_str = str(prev_val * curr_val)
    count += 1
  return count

File: c_t4_full/Clone_571.c
int main () {
    char a [] = {'2', '4', '6', '7', '7', '4', '4'};
    int i;
    for (i = 0; i < sizeof (a) / sizeof (char); i++) {
        my_set[a[i] - '0'].num++;
    }
    printf ("%-10s%-10s\n", "Value:", "Count:");
    for (i = 0; i < sizeof (my_set) / sizeof (TSet); i++) {
        if (my_set[i].num != 0) {
            printf ("%-10c%-10d\n", my_set [i].c, my_set [i].num);
        }
    }
}


 int main () {
    char a [] = {'2', '4', '6', '7', '7', '4', '4'};
    int i;
    char my_set[10];
    int j;

    for (i = 0; i < sizeof (a) / sizeof (char); i++) {
        for (j = 0; j < 10; j++) {
            if (my_set[j] == a[i]) {
                my_set[j + 10]++;
                break;
            } else if (my_set[j] == 0) {
                my_set[j] = a[i];
                my_set[j + 10]++;
                break;
            }
        }
    }
    printf ("%-10s%-10s\n", "Value:", "Count:");
    int k;
    for (i = 0; i < 10; i++) {
        k = 0;
        if (my_set[i] != 0) {
            for (j = 0; j < 10; j++) {
                if (my_set[j] == my_set[i]) {
                    k++;
                }
            }
            if (k != 0) {
                printf ("%-10c%-10d\n", my_set[i], k);
            }
        }
    }
}


